### PR TITLE
Add support for aes_256_gcm

### PR DESCRIPTION
**Pull request by trampster**
libre already had support, was just missing from baresip.

### CODE REVIEW

**Comment by alfredh** [comment]
many thanks !

can you please tell me how this was tested ?

File: modules/srtp/srtp.c [hunk 1]
```diff
@@ -34,8 +34,9 @@ struct menc_sess {
 struct menc_st {
 	/* one SRTP session per media line */
 	const struct menc_sess *sess;
-	uint8_t key_tx[32];
-	uint8_t key_rx[32];
+	uint8_t key_tx[32+12];
+	/* base64_decoding worst case encoded 32+12 key */
+	uint8_t key_rx[46];
```
Comment (alfredh) [inline modules/srtp/srtp.c]: is the comment correct? 32+12=44 and not 46 ?

Comment (trampster) [reply modules/srtp/srtp.c]: 32+12 is the size without base64 encoding, This is a result of the padding. we started with 44 and found out the hard way we needed 46.

**Comment by trampster** [comment]
Testing was performed as follows:

- We have a number of baresip based clients which have the preferred_suite (line 48 opf srtp.c) set to aes_256_gcm
- These talk to a Kamailio SIP server
- Media all goes to a central Conference Controller which handles the fanout for the group) this uses libre (not baresip) directly to do the aes_256_gcm
- We also have a client which bridges to a LMR (Land Mobile Radio) network, this also uses libre
- We have been running this for about a week here with about 5 units without issue.

What we haven't done:
- Tested against a non baresip/libre client
- Run a field trial

**Comment by alfredh** [comment]
Great work :)

**Pull request merged**